### PR TITLE
Legacy training option

### PR DESCRIPTION
**Pull request by hari-rangarajan**
Add an option to allow training using the COVER algorithm or the legacy algorithm

### CODE REVIEW

File: src/main/java/com/github/luben/zstd/Zstd.java [hunk 1]
```diff
@@ -303,21 +303,51 @@
...
+    /**
+     * Creates a new dictionary to tune a kind of samples (uses Cover algorithm)
+     *
+     * @param samples the samples buffer array
+     * @param dictBuffer the new dictionary buffer
+     * @return the number of bytes into buffer 'dictBuffer' or an error code if
+     *          it fails (which can be tested using ZSTD_isError())
+     */
+    public static long trainFromBuffer(byte[][] samples, byte[] dictBuffer) {
+        return trainFromBuffer(samples, dictBuffer, false);
```
Comment (hari-rangarajan) [inline src/main/java/com/github/luben/zstd/Zstd.java]: @luben I have these two overloaded methods for backward compatibility

File: src/main/java/com/github/luben/zstd/Zstd.java [hunk 2]
```diff
@@ -303,21 +303,51 @@
...
+     * Creates a new dictionary to tune a kind of samples (uses Cover algorithm)
+     *
+     * @param samples the samples direct byte buffer array
+     * @param sampleSizes java integer array of sizes
+     * @param dictBuffer the new dictionary buffer (preallocated direct byte buffer)
+     * @return the number of bytes into buffer 'dictBuffer' or an error code if
+     *          it fails (which can be tested using ZSTD_isError())
+     */
+    public static long trainFromBufferDirect(ByteBuffer samples, int[] sampleSizes, ByteBuffer dictBuffer) {
+        return trainFromBufferDirect(samples, sampleSizes, dictBuffer, false);
```
Comment (hari-rangarajan) [inline src/main/java/com/github/luben/zstd/Zstd.java]: @luben the other one

**Comment by luben** [comment]
Great, I missed them when reviewing. Merged

**Comment by luben** [comment]
@hari-rangarajan , just pushed 1.3.4-9 with the recent changes, it should arrive soon on maven central.

**Pull request merged**